### PR TITLE
Contractor changes for invoice resubmit

### LINKED ISSUE #2619: Contractor edits submitted Fixed-Price invoice, makes changes and resubmits it.
## Additional context
This is part of epic #2352.

## Describe the solution you'd like in Hypha - Design requirements

In Hypha the contractor:
- logs into after receiving an email notification from the PM
- reads the email requesting changes to a submitted invoice
- navigates to their project and chooses the invoice to be edited
- sees the invoice displayed and can edit the invoice, and make the required changes
	- these changes could be any/all of the following:
		- changing figures, dates in input fields
		- deleting uploaded invoice pdf file
		- uploading a new invoice pdf file
		- deleting uploaded supporting documents
		- uploading new supporting documents 
- can write a new message/comment
- saves the changes to the invoice

## Affected roles 
- Applicants

### Demo
Need to add.

### Wireframes

12.1 - The contrctor navigates to their project area and chooses the invoice to be edited. In this case invoice 09.

12.2 - The contractor sees the invoice displayed and can edit the invoice to make the required changes. These changes could be any/all of the following:
		- changing figures, dates in input fields
		- deleting uploaded invoice pdf file
		- uploading a new invoice pdf file
		- deleting uploaded supporting documents
		- uploading new supporting documents 

12.3 - here the contractor has changed the amount (from 3900.50 to 2900.50), they have also deleted the invoice pdf, and will upload another.

12.4 - here the contractor has uploaded a new invoice pdf file - project-name-Invoice-june-2021-correct.pdf

12.5 - The contractor has also written a new message for the OTF PM. The contractor then presses "submit changes" and the invoice has been submitted.

12.6 - The contractor now sees the invoice file embedded on the page. They can now close the browser window, or navigate back to the project area.

12.7 - The project page now shows the invoice with the resubmitted status. 

### What happens next?
The invoice will now be visible in the PM's list of invoices to review with the _resubmitted_ status.

### PR DESCRIPTION
**Pull request by chaitu210**
Fixes #2619

Add description here.

### CODE REVIEW

File: hypha/apply/projects/views/payment.py [hunk 1]
```diff
@@ -53,6 +56,21 @@ def get_form_kwargs(self):
 
     def form_valid(self, form):
         response = super().form_valid(form)
+        if form.cleaned_data['comment']:
+            invoice_status_change = _(f'<p>Invoice status updated to : {self.object.status_display}. </p>')
```
Comment (frjo) [inline hypha/apply/projects/views/payment.py]: F-strings and translations issue here. Use format().

File: hypha/apply/projects/views/payment.py [hunk 2]
```diff
@@ -137,6 +155,22 @@ def form_valid(self, form):
 
         response = super().form_valid(form)
 
+        if form.cleaned_data['message_for_pm']:
+            invoice_status_change = _(f'<p>Invoice status updated to : {self.object.status_display}. </p>')
```
Comment (frjo) [inline hypha/apply/projects/views/payment.py]: F-strings and translations issue here. Use format().

File: hypha/apply/projects/views/payment.py [hunk 3]
```diff
@@ -137,6 +155,22 @@ def form_valid(self, form):
...
+            message_for_pm = '<p>{}</p>'.format(form.cleaned_data['message_for_pm'])
```
Comment (frjo) [inline hypha/apply/projects/views/payment.py]: Prefer F-string here.

File: hypha/apply/projects/views/payment.py [hunk 4]
```diff
@@ -53,6 +56,21 @@ def get_form_kwargs(self):
 
     def form_valid(self, form):
         response = super().form_valid(form)
+        if form.cleaned_data['comment']:
+            invoice_status_change = _(f'<p>Invoice status updated to : {self.object.status_display}. </p>')
+            comment = '<p>{}</p>'.format(form.cleaned_data['comment'])
```
Comment (frjo) [inline hypha/apply/projects/views/payment.py]: Prefer F-string here.

File: hypha/apply/activity/messaging.py [hunk 5]
```diff
@@ -372,6 +373,14 @@ def handle_skipped_report(self, report, **kwargs):
         else:
             return "Marked a Report as required"
 
+    def handle_update_invoice_status(self, invoice, **kwargs):
+        invoice_status_change = _(f'Updated Invoice status to: {invoice.status_display}.')
+        return invoice_status_change
+
+    def handle_invoice_update(self, invoice, **kwargs):
+        invoice_status_change = _(f'Updated Invoice Info and Status changed to: {invoice.status_display}.')
```
Comment (frjo) [inline hypha/apply/activity/messaging.py]: F-strings and translations issue here. Use format().

File: hypha/apply/activity/messaging.py [hunk 6]
```diff
@@ -372,6 +373,14 @@ def handle_skipped_report(self, report, **kwargs):
         else:
             return "Marked a Report as required"
 
+    def handle_update_invoice_status(self, invoice, **kwargs):
+        invoice_status_change = _(f'Updated Invoice status to: {invoice.status_display}.')
```
Comment (frjo) [inline hypha/apply/activity/messaging.py]: F-strings and translations issue here. Use format().

File: hypha/apply/projects/views/payment.py [hunk 7]
```diff
@@ -162,22 +196,40 @@ def dispatch(self, request, *args, **kwargs):
...
+            messenger(
+                MESSAGES.UPDATE_INVOICE,
+                request=self.request,
+                user=self.request.user,
+                source=self.object.project,
+                related=self.object,
+            )
+
+            if 'message_for_pm' in form.changed_data:
+                invoice_status_change = _(f'<p>Invoice status updated to : {self.object.status_display}. </p>')
```
Comment (frjo) [inline hypha/apply/projects/views/payment.py]: F-strings and translations issue here. Use format().

File: hypha/apply/projects/views/payment.py [hunk 8]
```diff
@@ -162,22 +196,40 @@ def dispatch(self, request, *args, **kwargs):
...
+                message_for_pm = '<p>{}</p>'.format(form.cleaned_data['message_for_pm'])
```
Comment (frjo) [inline hypha/apply/projects/views/payment.py]: Prefer F-string here.

**Review by frjo — CHANGES_REQUESTED** [review]
Nice work! Just some translations issue.

P.S. I notice that there are a number of old translations issues in `messaging.py`. I will correct the mistakes (mine) in a separate PR.

**Review by frjo — APPROVED** [review]

**Comment by frjo** [comment]
Deployed to test just now.

File: hypha/apply/projects/views/payment.py (hunk 7 above)
Comment (frjo) [reply hypha/apply/projects/views/payment.py]: This seems to have been reverted by mistake.

File: hypha/apply/projects/views/payment.py (hunk 8 above)
Comment (frjo) [reply hypha/apply/projects/views/payment.py]: As have this.

**Review by frjo — CHANGES_REQUESTED** [review]

File: hypha/apply/projects/views/payment.py (hunk 7 above)
Comment (frjo) [reply hypha/apply/projects/views/payment.py]: While you are at it you can remove the extra space before the colon as well, here and in one or two other places as well.

Comment (frjo) [reply hypha/apply/projects/views/payment.py]: As well as space after the last period. It have no effect on the output but un neat stuff like that bothers me :-).

File: hypha/apply/projects/views/payment.py (hunk 3 above)
Comment (frjo) [reply hypha/apply/projects/views/payment.py]: Reverted here as well.

File: hypha/settings/dev.py [hunk 9]
```diff
@@ -143,8 +143,18 @@
 }
 
 WEBPACK_LOADER['DEFAULT'].update({
-    'STATS_FILE': os.path.join(BASE_DIR, './hypha/static_compiled/app/webpack-stats.json'),
+    'STATS_FILE': os.path.join(BASE_DIR, r'hypha\static_compiled\app\webpack-stats.json'),
```
Comment (frjo) [inline hypha/settings/dev.py]: This looks Windows specific?

File: hypha/settings/dev.py [hunk 10]
```diff
@@ -143,8 +143,18 @@
...
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': 'hypha',
+        'USER': 'postgres',
+        'PASSWORD': 'root',
+        'HOST': '127.0.0.1',
+        'PORT': '5432',
+    }
+}
```
Comment (frjo) [inline hypha/settings/dev.py]: This should go in to each developers local.py file I believe.

**Review by frjo — CHANGES_REQUESTED** [review]

**Review by sks444 — APPROVED** [review]
Looks good.

@frjo can we deploy this to test again? Thanks!

**Review by frjo — APPROVED** [review]

**Comment by frjo** [comment]
Deployed to test now.

**Comment by sks444** [comment]
@frjo do you think we could merge this in..if all looks good?

**Pull request merged**